### ISSUE TITLE
AZCopy 10 recursive from azfile share does not work

### ISSUE THREAD

**Issue by PaulCharlton** [Azure/azure-storage-azcopy#248]
### Which version of the AzCopy was used? 

10.0.8-preview

### Which platform are you using? (ex: Windows, Mac, Linux)

Mac

### What command did you run?
```
 azcopy cp "https://[account].file.core.windows.net/[share]/[path/to/directory]?[SAS]" "/path/to/dir" --recursive=true
```

### What problem was encountered?

documentation shows recursive copy from blob "folder"
```
Download an entire directory with SAS:
  - azcopy cp "https://[account].blob.core.windows.net/[container]/[path/to/directory]?[SAS]" "/path/to/dir" --recursive=true

Download files using wildcards with SAS:
  - azcopy cp "https://[account].blob.core.windows.net/[container]/foo*?[SAS]" "/path/to/dir"
```

expected same to work with file share
```
  - azcopy cp "https://[account].file.core.windows.net/[share]/[path/to/directory]?[SAS]" "/path/to/dir" --recursive=true
```
did not work, has error: ```failed to parse user input due to error: the inferred source/destination combination is currently not supported. Please post an issue on Github if support for this scenario is desired```

### How can we reproduce the problem in the simplest way?

create an azfile share with multiple files and directories, attempt to copy to local using the failed command `azcopy cp "https://[account].file.core.windows.net/[share]/[path/to/directory]?[SAS]" "/path/to/dir" --recursive=true`

### Have you found a mitigation/solution?

none found

**Comment by zezha-msft** [comment]
Hi @PaulCharlton, thanks for reaching out! It's awesome that you are already using 10.0.8, since I just released it 20 minutes ago!

I tried to repro the behavior you saw, but it downloads normally. Could you please double check that the URL is correct?

Btw what kind of SAS are you using? How did you generate it? Could you please try getting a URL (with SAS) from Storage Explorer and give it another try?

**Comment by zezha-msft** [comment]
One thing to note is that we expect the URl to be properly encoded. If you get the URL from Storage Explorer, they make sure to encode it for you.

**Comment by jorrite** [comment]
@zezha-msft, I can't reproduce the described scenario either. Downloading all files recursively to a local directory works fine. However, the scenario of copying files recursively from one file-share to another file-share does **not** work, and am getting the same `failed to parse user input due to error: the inferred source/destination combination is currently not supported. Please post an issue on Github if support for this scenario is desired` error as mentioned. Found this issue by searching on the error and thought it might be related so posting here instead of opening a new issue. Also worth noting that recursively copying from a file-share to a blob container does work. 

Steps to reproduce my issue:

```bash
end=`date -d "30 minutes" '+%Y-%m-%dT%H:%M:%SZ'`
SOURCE_SAS=$(az storage share generate-sas -n source --account-key <source_key> --account-name <source_name> --https-only --permissions dlrw --expiry $end -otsv)
DESTINATION_SAS=$(az storage share generate-sas -n destination --account-key < destination_name> --account-name <destination_name> --https-only --permissions dlrw --expiry $end -otsv)

azcopy cp "https://<source_name>.file.core.windows.net/source?$SOURCE_SAS" "https://<source_name>.file.core.windows.net/destination?$DESTINATION_SAS" --recursive=true
```

As mentioned, copying to a blob container works fine:
```bash
end=`date -d "30 minutes" '+%Y-%m-%dT%H:%M:%SZ'`
SOURCE_SAS=$(az storage share generate-sas -n source --account-key <source_key> --account-name <source_name> --https-only --permissions dlrw --expiry $end -otsv)
DESTINATION_SAS=$(az storage container generate-sas -n destination --account-key < destination_name> --account-name <destination_name> --https-only --permissions dlrw --expiry $end -otsv)

azcopy cp "https://<source_name>.file.core.windows.net/source?$SOURCE_SAS" "https://<source_name>.blob.core.windows.net/destination?$DESTINATION_SAS" --recursive=true
```

Before running the above 2 snippets, I made sure source (and destination) exist and added some files/folders to the source file-share.

**Comment by zezha-msft** [comment]
Hi @jorrite, glad to hear that the original scenario is unblocked for you.

File share -> file share is currently not supported, due to service side limitations. You could potentially map in the destination drive onto your system, and perform a download from File share -> local.

Please let us know if you have any other question.

**Comment by Porwal7** [comment]
Hi @zezha-msft , not able to copy data from between subscription .
getting an issue "the inferred source/destination combination is currently not supported. Please post an issue on Github if support for this scenario is desired"

./azcopy cp "https://[account].dfs.core.windows.net/[container]/[path/to/directory]"  "https://[account].dfs.core.windows.net/[container]/[path/to/directory]"

**Comment by zezha-msft** [comment]
Hi @Porwal7, sorry for the inconvenience. This scenario is not yet supported due to service limitations.

**Comment by judgewest2000** [comment]
I'm having the exact same problem

**Comment by freeza1990** [comment]
> I'm having the exact same problem



bro, azcopy for blob does not copy empty files, because blob is an object store, not a file store. ie when the folder is empty and as if it did not exist in blob because it does not have the same structure as a fileserver or system for example.

**Comment by JohnRusk** [comment]
@freeza1990 I think you meant to say "azcopy for blob does not copy empty _folders_".  That's absolutely correct, for exactly the reason you describe.  We _do_ copy empty _files_.

**Comment by JohnRusk** [comment]
@judgewest2000 did @freeza1990 's reply answer your question?  (That there is no such thing as an empty folder in blob storage, because folders in blob storage are just an illusion https://stackoverflow.com/a/2619111).
